controllers/apps/transformer_cluster_credential.go

```diff
@@ -47,9 +49,23 @@ func (c *ClusterCredentialTransformer) Transform(ctx graph.TransformContext, dag
 		return err
 	}
```

Comment:
expect to have cluster componentSpecs validation where 2 or more frontend components is not allowed

Reply:
done